File: SEGUNDO/C1/NetBeansProjects/Deep_space/src/deepspace/Hangar.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author daniel
 */
public class Hangar {
    private int maxElements;
    private ArrayList<ShieldBooster> shieldBoosters;
    private ArrayList<Weapon> weapons;
    
    public Hangar(int capacity){
        this.maxElements=capacity;
    }
    
    public Hangar(Hangar h){
        this.maxElements=h.getMaxElements();
        this.shieldBoosters=(ArrayList<ShieldBooster>) h.getShieldBoosters().clone();
        this.weapons=(ArrayList<Weapon>) h.getWeapons().clone();
    }
    
    public HangarToUI getUIversion(){
        return new HangarToUI(this);
    }
    
    private boolean spaceAvailable(){
        boolean salida;
        if((shieldBoosters.size() + weapons.size()) < maxElements)
            salida=true;
        
        else
            salida=false;
        
        return salida;
    }
    
    public boolean addWeapon(Weapon w){
        boolean salida;
        
        if(this.spaceAvailable()){
            weapons.add(w);
            salida=true;
        }
        
        else
            salida=false;
        
        return salida;
    }
    
    public boolean addShieldBooster(ShieldBooster s){
        boolean salida;
        
        if(this.spaceAvailable()){
            shieldBoosters.add(s);
            salida=true;
        }
        
        else
            salida = false;
        
        return salida;
    }
    
    public int getMaxElements(){
        return this.maxElements;
    }
    
    public final ArrayList<ShieldBooster> getShieldBoosters(){
        return this.shieldBoosters;
    }
    
    public final ArrayList<Weapon> getWeapons(){
        return this.weapons;
    }
    
    public ShieldBooster removeShieldBooster(int s){
        ShieldBooster salida;
        
        if( !shieldBoosters.isEmpty() && s < shieldBoosters.size())
            salida=shieldBoosters.remove(s);
        
        
        else
            salida=null; 
        
        return salida;
    }
    
    public Weapon removeWeapon(int w){
        Weapon salida;
        
        if( !weapons.isEmpty() && w < weapons.size())
            salida=weapons.remove(w);
        
        else
            salida=null;
        
        return salida;   
    }
    
}

File: SEGUNDO/C2/PDOO/backup/DeepSpace_ruby/lib/Hangar.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
#encoding:utf-8
require_relative 'HangarToUI'
require_relative 'Weapon'
require_relative 'ShieldBooster'
require_relative 'WeaponType'

module Deepspace
class Hangar
  def initialize(capacity)
    @maxElements=capacity
    @weapons=Array.new
    @shieldBoosters=Array.new
  end
  
  def self.newCopy(h)
    new(h.maxElements)
  end
  
  def getUIversion
    return HangarToUI.new(self)
  end
  
  def spaceAvailable
    return ((@weapons.length + @shieldBoosters.length) < @maxElements)
    
  end
  
  def addWeapon(w)
    salida=spaceAvailable
    
    if(salida)
      @weapons.push(w)
    end
    
    return salida    
  end
  
  def addShieldBoosters(s)
    salida=spaceAvailable
    
    if(salida)
      @shieldBoosters.push(s)
    end
    
    return salida    
  end
  
  def maxElements
    @maxElements
  end
  
  def shieldBoosters
    @shieldBoosters
  end
  
  def weapons
    @weapons
  end
  
  def removeShieldBoosters(s)
    return @shieldBoosters.delete_at(s)
    
  end
  
  def removeWeapon(w)
    return @weapons.delete_at(w)
  end
  
  def to_s
    salida="Elementos: #{@maxElements}\n"
    
    if(@weapons != nil)
      salida=salida + @weapons.join("\n")
    end
    
    if(@shieldBoosters != nil)
      salida=salida+ "\n" + @shieldBoosters.join("\n")
    end
    
    return salida
    
  end
  
  private :spaceAvailable
end

end
